_episodes/06-debugging.md

```diff
@@ -0,0 +1,309 @@
+they become much easier to fix.
+The good news is, ``ESMValTool`` creates a record of the output messages and stores them in log files.
+They can be used for debugging or monitoring the process.
+This lesson helps to understand what the different types of errors are and when you are likely to encounter them.
+
+## log files
+
+Each time we run the ESMValTool, it will produce a new output directory.
+This directory should contain the ``run`` folder that is automatically generated by ESMValTool.
+To examine this, we run an example recipe that can be found in:
+[recipe_example.yml](link).
+Let's download it to our working directory ``esmvaltool_tutorial`` that was created during
+the [Setup]({{ page.root }}{% link setup.md %}).
+
+In a new terminal, go to our working directory ``esmvaltool_tutorial`` where
+both files ``user-config.yml`` and ``recipe_example.yml`` are located and run the recipe:
+
+~~~bash
+  cd esmvaltool_tutorial
+  esmvaltool -c user-config.yml recipe_example.yml
+~~~
+
+~~~
+esmvaltool: command not found
+~~~
+{: .error}
+
+ESMValTool encounters this error because
+the conda environment ``esmvaltool`` has not been activated.
+To fix the error, before running the recipe, activate the environment:
+
+~~~bash
+conda activate esmvaltool
+~~~
+
+> ## conda environment
+>
+> More information about conda environment can be found at
+[Installation]({{ page.root }}{% link _episodes/02-installation.md %})
+{: .callout}
+
+Let's change the working directory to the folder ``run`` and list its files:
+
+~~~bash
+  cd path_to_output_directory/run
+  ls
+~~~
+
+~~~
+main_log_debug.txt   main_log.txt  recipe_example.yml   resource_usage.txt
+~~~
+{: .output}
+
+In the ``main_log_debug.txt`` and ``main_log.txt``, ESMValTool tells us the output messages,
+warnings and possible errors that might happen during pre-processings.
+To inspect them, we can look inside the files. For example:
+
+~~~bash
+  cat main_log.txt
+~~~
+
+If you encounter an error and don’t know what it means, it is important to read the log information.
+Sometimes knowing where the error occurred is enough to fix it, even if you don’t entirely understand the message.
+However, note that you may not always be able to find the error or fix it.
+In that case, ESMValTool community helps you figure out what went wrong.
+
+> ## Different log files
+>
+> In the ``run`` directory, there are two log files ``main_log_debug.txt`` and ``main_log.txt``.
+What are their differences?
+>
+>> ## Solution
+>>
+>> The ``main_log_debug.txt`` contains the output messages from the pr-processors whereas
+the ``main_log.txt`` shows general errors and warnings that might happen in running the recipe and diagnostics script.
+> {: .solution}
+{: .challenge}
+
+Let's change some settings in the recipe to run a regional pre-processor.
+We run a text editor called ``Nano`` to open the recipe file:
+
+~~~bash
+  cd esmvaltool_tutorial
```

Comment:
it works most of the time. Also, this command reminds the user to change the working directory.

Reply:
I'm asking because just above, you instruct to do `cd path_to_output_directory/run/diag_timeseries_temperature`, so I guess here you'll need `cd ../../../` to get back, no? Maybe just type "`cd` back into the directory `esmvaltool_tutorial`"?